src/pages/SingleVideo/SingleVideo.js

```diff
@@ -1,7 +1,57 @@
 import React from "react";
+import { useParams } from "react-router-dom";
+import { useVideos } from "../../context/videos-context";
+import "./singleVid.css";
 
 const SingleVideo = () => {
-	return <div>SingleVideo</div>;
+	const { videoId } = useParams();
+	const { videos } = useVideos();
+	const getSingleVideo = (videos, videoId) => {
+		return videos.find((video) => video.id === videoId);
+	};
+	const video = getSingleVideo(videos, videoId);
+
+	return (
+		<>
+			<div className="single-vid-card">
+				<div className="video-frame">
+					<iframe
+						width="100%"
+						height="100%"
+						src={`https://www.youtube.com/embed/${video.id} `}
+						frameBorder="0"
+						allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
+						allowFullScreen=""
```

Comment:
You can avoid using inline styles 
Also please note that you can also use reactplayer(can read about it online) as it is better alternative than iframe

Reply:
Thank you Samyak for suggestion. Sure will look into it